### PR TITLE
Ignore NaN values in wg_typed_value_create_from_value_t_inline.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Should get rid of the "can not take infinite value" log spam.

### CODE REVIEW

**Review by davidbtucker — COMMENTED** [review]
Quick question: Any unit tests to update for this PR?

File: src/write_gcm.c [hunk 1]
```diff
@@ -1528,8 +1528,9 @@ static int wg_typed_value_create_from_value_t_inline(wg_typed_value_t *result,
   switch (ds_type) {
     case DS_TYPE_GAUGE: {
       if (!isfinite(value.gauge)) {
-        ERROR("write_gcm: can not take infinite value");
-        return -1;
+        DEBUG("write_gcm: ignoring a NaN or infinite value");
+        result->field_name_static = NULL;
+        return 0;
```
Comment (davidbtucker) [inline src/write_gcm.c:1530]: Hm, I don't really love the pattern of "return success but the value may be invalid so the caller better remember to check its contents too".

Would it be possible to return a more nuanced status than just 0/-1 here? Actually, even returning -2 and having the caller check for that value explicitly would be less error-prone I think:

    int status = wg_typed_value_create_from_value_t_inline(&typed_value,
        value->ds_type, value->val, &data_source_type_static) != 0
    if (status != 0) {
      if (status == -2) {
        DEBUG("write_gcm: wg_typed_value_create_from_value_t_inline ignored "
              "%s/%s/%s. Continuing.",
              element->key.plugin, element->key.type, value->name);
      } else {
        WARNING("write_gcm: wg_typed_value_create_from_value_t_inline failed "
                "for %s/%s/%s! Continuing.",
                element->key.plugin, element->key.type, value->name);
      }
      continue;
    }

What do you think?

**Review by davidbtucker — APPROVED** [review]
Approved.

Left one suggestion, feel free to ignore as desired.

File: src/write_gcm.c (hunk 1 above)
Comment (igorpeshansky) [reply src/write_gcm.c:1530]: I've actually initially considered doing it the way you're suggesting, but decided against it. The current implementation effectively makes `NULL` a valid value that means "ignore". Mixing up the valid case (explicit ignore) and the invalid case (some error occurred) rubs me the wrong way a bit.
I did add a comment to clarify the new behavior. You've also made me realize that the whole interface to that function can be simplified, so I pushed another commit for that. Take a look and see if it addresses your concerns.

**Review by davidbtucker — APPROVED** [review]

**Pull request merged**